lib/compress/zstd_opt.c

```diff
@@ -818,6 +964,38 @@ ZSTD_compressBlock_opt_generic(ZSTD_matchState_t* ms,
     ZSTD_match_t* const matches = optStatePtr->matchTable;
     ZSTD_optimal_t lastSequence;
 
+    U32 ldmStartPosInBlock = 0;
+    U32 ldmEndPosInBlock = 0;
+    U32 ldmOffset = 0;
+    
```

Comment:
@terrelln So this comment below goes into more explicit detail about the surprising behavior I mentioned during our 1:1 and the code below is the my supposed "mitigation", basically pushing back adding the match by adding more bytes to the litLength of the ldm.

Reply:
@senhuang42 I see the problem now. This is not the correct solution. This will likely break on large files.

I believe this is caused by `btultra2` which does 2 passes over the first block. You can double check by seeing if the problem goes away at level 18, which uses `btultra`, not `btultra2`. ` You'll need to reset the LDM seqStore after the first pass, to re-consume the sequences.

https://github.com/facebook/zstd/blob/cc88eb7594d9c70ec2440ce6122f7861dbd64af2/lib/compress/zstd_opt.c#L1123-L1129